Alignment/OfflineValidation/interface/SmartSelectionMonitor.h

```diff
@@ -0,0 +1,133 @@
+#ifndef TKALTOOLS_JETHTANALYZER_SMARTSELECTIONMONITOR_H
+#define TKALTOOLS_JETHTANALYZER_SMARTSELECTIONMONITOR_H
+
+// system include files
+#include <iostream>
+#include <string>
+#include <map>
+#include <algorithm>
+#include <vector>
+#include <memory>
+#include <unordered_map> 
+
+// user include files
+#include "TH1D.h"
+#include "TH2D.h"
+#include "TProfile.h"
+#include "TString.h"
+#include "TROOT.h"
+
+#include <ext/hash_map>
+
+namespace __gnu_cxx{
+  template<> struct hash< TString >{ size_t operator()( const TString& x ) const{ return hash< const char* >()( x.Data() );  }  };
+}
+
+class SmartSelectionMonitor {
+  
+public:
+
+  SmartSelectionMonitor(){}
+  ~SmartSelectionMonitor() { }
+
+  typedef __gnu_cxx::hash_map<TString, std::map<TString, TH1*>* > Monitor_t;
```

Comment:
I think in the meanwhile this type of container is deprecated in the master of cmssw, shall we try with an STL unordered map?

Reply:
Yeah, while hash_map works, it gives a deprecation warning during compilation. However, if I just change hash_map to unordered_map, it compiles ok, but none of the histograms get filled. I am not sure what else is needed to make the code work using unordered_map.

Reply:
I found the reason why unordered_map did not work. The code was using TString as a key for the unordered_map and the equality operator for TString is not working as one would expect when comparing strings. Namely, if TString one = "myString" and TString two = "myString", (one == two) returns false. This breaks the find-function of the unordered_map and thus the code thought none of the keys it tried to seek existed in the map. The fix was to replace TString with std::string as a key value. After this and some necessary adjustments elsewhere to the code, it now compiles and works without giving the deprecation warning.

Reply:
Excellent, thank you